src/components/atoms/AddDatatoken.tsx

```diff
@@ -0,0 +1,30 @@
+import React, { ReactElement } from 'react'
+import Button from './Button'
+import { addTokenToWallet } from '../../utils/web3'
+import { useWeb3 } from '../../providers/Web3'
+
+export default function AddToken({
+  address,
+  symbol
+}: {
+  address: string
+  symbol: string
+}): ReactElement {
+  const { web3Provider } = useWeb3()
+
+  async function handleAddToken() {
+    await addTokenToWallet(
+      web3Provider,
+      address,
+      // TODO: figure out how to add full symbol
```

Comment:
this is actually a problem. There is no restriction on the length of a ERC-20 token symbol, but more than 6 is very uncommon and it looks like MetaMask opted to cap it there. But the `-13` numbers are part of the symbol so they need to be there. Not sure we can solve this apart from asking MetaMask to not be so picky

Reply:
One other thing we could do (but doesn't help the existing datatokens) is to change our function for creating the symbol on the publish page, so that it's limited to 6 characters

Reply:
https://github.com/MetaMask/metamask-extension/issues/9243

https://github.com/MetaMask/metamask-extension/pull/5816

Reply:
Yeah, I got the same error. That's why I implemented it as a substring of 6 characters

Reply:
I just had a look at the code and it is still set to 12 characters: https://github.com/jamiehewitt15/metamask-extension/blob/bc24d9e696560780c13c94a39393cd8b6bf16147/ui/pages/add-token/add-token.component.js#L206

I'll open an issue on the Metamask GitHub and see if we can get any response

Reply:
I've created an issue, hopefully we can get some help on this: https://github.com/MetaMask/metamask-extension/issues/10973

Reply:
hard to figure out their exact release process but from what I'm seeing the fix they already have in `develop` is not in latest 9.4.0 release https://github.com/MetaMask/metamask-extension/blob/v9.4.0/app/scripts/controllers/preferences.js#L823-L827

So guess we have to wait for a new MetaMask release